TODO

```diff
@@ -118,14 +118,20 @@ Features:
 
 * seccomp: maybe merge all filters we install into one with that libseccomp API that allows merging.
 
-* per-service credential system. Specifically: add LoadCredential= (for loading
-  cred from file), AcquireCredential= (for asking user for cred, via
-  ask-password), PassCredential= (for passing on credential systemd itself
-  got). Then, place credentials in a per-service, immutable ramfs instance (so
-  that it cannot be swapped out), destroy after use. Also pass via keyring
-  (with graceful fallback to cover for containers). Define CredentialPath= for
-  defining subdir of /run/credentials/ where to place it. Set $CREDENTIAL_PATH
-  env var for services to the result. Also pass via fd passing (optionally).
+* credentials system:
+  - maybe add AcquireCredential= for querying a cred via ask-password
+  - maybe try to acquire creds via keyring?
+  - maybe try to pass creds via keyring?
+  - maybe optionally pass creds via memfd
+  - maybe add support for decrypting creds via TPM
```

Comment:
how about decrypting creds via `pkcs11` too?  We now have the code to talk to pkcs11 hsm's now anyway due to homed

Do we have any code in systemd regarding TPM crypto yet? Or only measurement? I'm a bit fuzzy on using TPM for cryptography usecase

Reply:
We only do some TPM stuff in sd-boot. It's high on my TODO list to add some support to systemd-cryptsetup though, matching the existing pkcs11 support.

and yeah I guess we could decode creds via pkcs11 too. But pkcs11 is so generic I have the suspicion that people might want to use it in entirely different ways: i.e. some would just want to import keys from pkcs11 into the file system. Others might want to decrypt host-stored keys with keys stored on pkcs11. Others might want to HMAC some data stored in the fs with some key on the pkcs11 device. There are so many ways one might want to to use pkcs11, and I am not sure what the right approach there is...

Also, pkcs11 might be interactive, at which point it might be a less than ideal choice for this kind of stuff...

Reply:
anyway, will add to TODO list too.